src/Tribe/Repository/Decorator.php

```diff
@@ -592,4 +592,56 @@ public function where_multi( array $fields, $compare, $value, $where_relation =
 
 		return $this;
 	}
+
+	/**
+	 * Handle getting additional property from decorated object.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $name Property name.
+	 *
+	 * @return mixed
+	 */
+	public function __get( $name ) {
+		return $this->decorated->{$name};
+	}
+
+	/**
+	 * Handle setting additional property on decorated object.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $name  Property name.
+	 * @param mixed  $value Property value.
+	 */
+	public function __set( $name, $value ) {
+		$this->decorated->{$name} = $value;
+	}
+
+	/**
+	 * Check if additional property on decorated object exists.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $name Property name.
+	 *
+	 * @return bool
+	 */
+	public function __isset( $name ) {
+		return isset( $this->decorated->{$name} );
+	}
+
+	/**
+	 * Call methods on decorated object.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $name      Method name.
+	 * @param array  $arguments Method arguments.
+	 *
+	 * @return mixed
+	 */
+	public function __call( $name, $arguments ) {
+		return call_user_func_array( [ $this->decorated, $name ], $arguments );
```

Comment:
I think an exception might be good if `this->decorated` is not `callable` before `call_user_func_array`

Reply:
I'm not sure I want to make that sort of decision at this point. It's not the duty of this class to add exceptions where none are expected, it will cause a fatal either way.

Reply:
I grabbed some of this code from ical-tec Events repository, it had implemented it on it's own apart from the Decorator in common. It didn't check if a property exists before trying to call it. I think it's important to remain neutral on whether the method/property exists or is public.

Reply:
I see, however if `$this->decorated` is not callable an error still will be trigger by `call_user_func_array`. I suggest an exception as a way to communicate more clear but if this is the current design on `common` probably just a check before the call will be enough, however when `decorated` is not callable will fail quietly

Reply:
Lets go with how @sc0ttkclark  did and in the future adjust.

Reply:
Yeah not a major stuff can be moved forward for now…